controllers/articles.controller.js

```diff
@@ -16,11 +17,25 @@ exports.getArticlesById = (request, response, next) => {
 };
 
 exports.getArticles = (request, response, next) => {
-  selectArticles()
-    .then((articles) => {
-      response.status(200).send({ articles });
-    })
-    .catch(next);
+  const { topic, sort_by, order } = request.query;
+  if (topic) {
+    const promises = [
+      selectArticleQueries(topic, sort_by, order),
+      checkExists("articles", "topic", topic),
+    ];
+    Promise.all(promises)
+      .then((resolvedPromises) => {
+        const articles = resolvedPromises[0];
+        response.status(200).send({ articles });
+      })
+      .catch(next);
+  } else {
```

Comment:
I'd like to see this conditional checking of query parameters in the model rather than the controller